### PR TITLE
Allow specifying versions of articles

### PR DESCRIPTION
**Pull request by wisygig**
In this PR we address #3:
* Added `version` parameter to `get_article_details` and `list_files`.
* Fixed an error in handling HTTPError exceptions.

### CODE REVIEW

**Comment by vasudevanv** [comment]
I am not sure if the Figshare API supports getting details for a specific version of a private article. 
Maybe we should just throw an exception for that case.

File: figshare/figshare.py [hunk 1]
```diff
@@ -1,8 +1,4 @@
-import sys
-import os
-import hashlib
 import json
-
```
Comment (dhimmel) [inline figshare/figshare.py:5]: Let's keep this newline but delete the gap between `import requests` and `from requests.exceptions import HTTPError`. As per PEP8

File: figshare/figshare.py [hunk 2]
```diff
@@ -181,33 +180,45 @@ def update_article(self, article_id, **kwargs):
...
 
-    def list_files(self, article_id):
-        """ List all the files associated with a given article.
+    def list_article_versions(self, article_id):
+        """ Return the details of an article with a given article ID.
 
         Parameters
         ----------
-        article_id : str or int
+        article_id : str or id
```
Comment (dhimmel) [inline figshare/figshare.py]: Keep as int. This part of the docstring specifies the type

**Comment by dhimmel** [comment]
@wisygig just checking in. This is an important PR!

**Comment by wisygig** [comment]
@dhimmel I'll finish this up tonight at the meetup.

**Review by dhimmel — APPROVED** [review]
Thanks for adding this important functionality!

**Pull request merged**